### PR TITLE
Helm Chart to have the "meshery" namespace

### LINKED ISSUE #7893: [CI] Enhance Meshery's Helm Chart to have the "meshery" namespace included
#### Current Behavior
The current Meshery Helm chart excludes definition of a namespace in which to provision Meshery.

#### Desired Behavior
Include default namespace, named `meshery`.

#### Acceptance Tests
1. the default "Meshery" namespace should be overridable by a setting in values.yaml.

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Freedisch**
**Notes for Reviewers**

This PR fixes #7893 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
LGTM, how about you, @AugustasV?

**Review by leecalcote — DISMISSED** [review]

**Comment by leecalcote** [comment]
@AugustasV, speaking of things that look good to you, how does this look? https://github.com/meshery/meshery/pull/8044#issuecomment-1619068573

**Comment by AugustasV** [comment]
@Freedisch why did you do that? any comment? Not cool to do do like that

**Comment by Freedisch** [comment]
@AugustasV There is the Release chart ci failing, it seems metadata.namespace field is not directly accessible in the deployment manifest

File: install/kubernetes/helm/meshery/templates/deployment.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ include "meshery.name" . }}
-  namespace: {{ .Release.Namespace }}
+  namespace: {{ .Values.namespace }}
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply install/kubernetes/helm/meshery/templates/deployment.yaml]: Good one, @AugustasV 👏

**Review by Abhishek-kumar09 — COMMENTED** [review]
We don't require changes in the kubernetes manifests right?

File: install/kubernetes/helm/meshery/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 apiVersion: v2
 name: meshery
 description: Meshery chart for deploying Meshery and Meshery's adapters.
+metadata:
```
Comment (MUzairS15) [inline install/kubernetes/helm/meshery/Chart.yaml:4]: I think it is more appropriate to have it under Values.yaml.

**Comment by Chadha93** [comment]
@Freedisch Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Review by leecalcote — CHANGES_REQUESTED** [review]
@Freedisch @AugustasV, here is what I get while running `make helm-lint` against these changes:

```
==> Linting install/kubernetes/helm/meshery
[ERROR] templates/: template: meshery/templates/deployment.yaml:5:25: executing "meshery/templates/deployment.yaml" at <.metadata.namespace>: nil pointer evaluating interface {}.namespace
```

**Comment by Freedisch** [comment]
> @Freedisch @AugustasV, here is what I get while running `make helm-lint` against these changes:
> 
> ```
> ==> Linting install/kubernetes/helm/meshery
> [ERROR] templates/: template: meshery/templates/deployment.yaml:5:25: executing "meshery/templates/deployment.yaml" at <.metadata.namespace>: nil pointer evaluating interface {}.namespace
> ```

I'll update it, I think `{{metadata.namespace}}` is not accessible...👀 @AugustasV

File: install/kubernetes/helm/meshery/values.yaml [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 replicaCount: 1
 restartPolicy: Always
+namespace: meshery
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply install/kubernetes/helm/meshery/values.yaml]: Does this act as a default setting? Without it, will `meshery` be the default namespace, @AugustasV?

**Pull request merged**